File: jdroid-java-core/src/main/java/com/jdroid/java/collections/Iterators.java
package com.jdroid.java.collections;

import java.util.Iterator;

public class Iterators {
	
	/**
	 * Returns the number of elements remaining in {@code iterator}. The iterator will be left exhausted: its
	 * {@code hasNext()} method will return {@code false}.
	 * 
	 * @param iterator The {@link Iterator}
	 * @return The {@link Iterator} size
	 */
	public static int size(Iterator<?> iterator) {
		int count = 0;
		while (iterator.hasNext()) {
			iterator.next();
			count++;
		}
		return count;
	}
}

File: jdroid-java-core/src/main/java/com/dipien/java/collections/Iterators.kt
package com.dipien.java.collections

object Iterators {

    /**
     * Returns the number of elements remaining in `iterator`. The iterator will be left exhausted: its
     * `hasNext()` method will return `false`.
     *
     * @param iterator The [Iterator]
     * @return The [Iterator] size
     */
    fun size(iterator: Iterator<*>): Int {
        var count = 0
        while (iterator.hasNext()) {
            iterator.next()
            count++
        }
        return count
    }
}
